marlowe-dashboard-client/src/Capability/Wallet.purs

```diff
@@ -0,0 +1,82 @@
+module Capability.Wallet
+  ( class MonadWallet
+  , createWallet
+  , submitWalletTransaction
+  , getWalletPubKey
+  , updateWalletPaymentWithChange
+  , getWalletSlot
+  , getWalletTransactions
+  , getWalletTotalFunds
+  , signTransaction
+  ) where
+
+import Prelude
+import AppM (AppM)
+import Bridge (toBack, toFront)
+import Capability.Ajax (runAjax)
+import Control.Monad.Except (lift)
+import Data.Json.JsonTuple (JsonTuple)
+import Data.Map (Map)
+import Data.Newtype (unwrap)
+import Halogen (HalogenM)
+import MainFrame.Types (WebData)
+import Marlowe.Semantics (Assets, PubKey, Slot)
+import Plutus.PAB.Webserver (getWalletByWalletIdOwnoutputs, getWalletByWalletIdOwnpublickey, getWalletByWalletIdTotalfunds, getWalletByWalletIdWalletslot, postWalletByWalletIdSign, postWalletByWalletIdSubmittxn, postWalletByWalletIdUpdatepaymentwithchange, postWalletCreate)
+import Plutus.V1.Ledger.Tx (Tx, TxOutRef, TxOutTx)
+import Wallet.Types (Payment)
+import WalletData.Types (Wallet)
+
+-- The PAB PSGenerator (using Servant.PureScript) automatically generates a PureScript module with
+-- functions for calling all Wallet API endpoints. This `MonadWallet` class wraps these up in a
+-- 'capability' monad (https://thomashoneyman.com/guides/real-world-halogen/push-effects-to-the-edges/)
+-- with some nicer names and type signatures, mapping the result to WebData.
+class
+  Monad m <= MonadWallet m where
+  createWallet :: m (WebData Wallet)
+  submitWalletTransaction :: Tx -> Wallet -> m (WebData Unit)
```

Comment:
Whate are wallet transactions? I only knew about contract transaction

Reply:
I don't know. Like the `Contract` capability, this is intended to be a low level wrapper around the automatically generated wallet API module (ultimately destined for a web-common directory). I think we only need three functions here: `createWallet`, `getWalletPubKey` and `getWalletTotalFunds`.